src/UI/SlidingValue.java

```diff
@@ -9,84 +9,73 @@
 public class SlidingValue
 {
   private static final long UPDATE_DELAY_MS = 16; // Minimum time to update the intermediate position.
-  private int actualValue;
-  private double visualValue;
+  private int targetValue;
+  private double currentValue;
   private long lastTime;
-  private long unusedTime;
 
   public SlidingValue(int val)
   {
-    actualValue = val;
-    visualValue = val;
+    targetValue = val;
+    currentValue = val;
     lastTime = System.currentTimeMillis();
   }
 
   /** Set a target/destination for this SlidingPoint. */
   public void set(int val)
   {
-    if( val != actualValue )
+    if( val != targetValue )
     {
-      actualValue = val;
+      targetValue = val;
       lastTime = System.currentTimeMillis();
     }
   }
 
   /** Set a target/destination for this SlidingPoint and move it there instantly. */
   public void snap(int val)
   {
-    actualValue = val;
-    visualValue = val;
+    targetValue = val;
+    currentValue = val;
     lastTime = System.currentTimeMillis();
   }
 
   /** Retrieve the "current" location of the point as it slides to its destination. */
   public double get()
   {
-    if( visualValue != actualValue )
+    // If we are already basically there, just snap to.
+    double snapDistance = 0.05;
+    if( Math.abs(targetValue - currentValue) < snapDistance )
     {
-      double slide = calculateSlideAmount(visualValue, actualValue);
-      visualValue += slide;
+      currentValue = targetValue;
     }
-    return visualValue;
+
+    // Otherwise, figure out how much closer to get and move in.
+    if( currentValue != targetValue )
+    {
+      double slide = calculateSlideAmount();
+      currentValue += slide;
+    }
+    return currentValue;
   }
```

Comment:
Seeing as my implementation is simpler still than this simplification, I think I'd still like to see mine happen.

If you're dead set on having "snap to" logic (which isn't really needed IMO), just keep a bool to record whether you've arrived yet, and just return the destination without any other logic needed if it's true.

Reply:
I like "snap to" because it prevents tons of unnecessary calculations and the single-pixel shift 1.5 seconds after you thought you were already there.